File: I. EASY/242. Valid Anagram/solution.cpp
class Solution
{
public:
    bool isAnagram(string s, string t)
    {
        int ascii[256];

        for (int i = 0; i < 256; i++)
            ascii[i] = 0;

        for (int i = 0; i < s.size(); i++)
            ascii[s[i]]++;

        for (int i = 0; i < t.size(); i++)
            ascii[t[i]]--;

        for (int i = 0; i < 256; i++)
            if (ascii[i] != 0)
                return false;

        return true;
    }
};

File: I. EASY/242. Valid Anagram/solution.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        hmap = {}

        # Add all characters' frequencies in s
        for c in s:
            if c in hmap:
                hmap[c] += 1
            else:
                hmap[c] = 1

        # Subtract all characters' frequencies in t
        for c in t:
            if c in hmap:
                hmap[c] -= 1
            else:
                return False

        # Check if all characters' frequencies are 0
        # If not, it means that the strings are not anagrams
        for c in hmap:
            if hmap[c] != 0:
                return False

        # If all characters' frequencies are 0, it means that the strings are anagrams
        return True
